src/dotnet/Program.cs

```diff
@@ -113,6 +113,16 @@ private static int ProcessArgs(string[] args)
 
             var appArgs = (lastArg + 1) >= args.Length ? Enumerable.Empty<string>() : args.Skip(lastArg + 1).ToArray();
 
+            if (verbose.HasValue)
+            {
+                Environment.SetEnvironmentVariable(CommandContext.Variables.Verbose, verbose.ToString());
+            }
+            //This process should inherit AnsiPassThrough or use the default of false.
+            CommandContext.InitializeAnsiPassThroughFromEnvironment();
```

Comment:
I have an incoming PR which may overlap a bit but it shouldn't be too bad. :smile: 

I get what you are trying to do now, but the way of doing this feels hacky. 
Reading through this bit it wouldn't be clear to me that we are saving the value so that it's not affected by the subsequent set.

With your other change this should be entirely unnecessary for internal commands now no?
So why not only set this environment variable when calling external commands, leaving it in the command.create below but removing it altogether for internal commands.